cvat/apps/engine/serializers.py

```diff
@@ -1163,37 +1188,61 @@ def _to_rel_frame(abs_frame: int) -> int:
                                     chunk_id,
                                     chunk_frames,
                                     quality,
-                                    {chunk_frame: all_task_frames[chunk_frame].path for chunk_frame in chunk_frames},
+                                    {
+                                        chunk_frame: all_task_frames[chunk_frame].path
+                                        for chunk_frame in chunk_frames
+                                    },
                                     segment_frame_map,
                                 ],
                             ),
                         )
                         wait_for_rq_job(rq_job)
 
-                    MediaCache().remove_segment_chunk(db_segment, chunk_id, quality=quality)
+                    chunks_to_be_removed.append(
+                        {'db_segment': db_segment, 'chunk_number': chunk_id, 'quality': quality}
+                    )
 
-            db_segment.chunks_updated_date = timezone.now()
-            db_segment.save(update_fields=['chunks_updated_date'])
+            if bulk_context:
+                media_cache = bulk_context.media_cache
+            else:
+                media_cache = MediaCache()
+            media_cache.remove_segment_chunks(chunks_to_be_removed)
```

Comment:
Why do we need to intruduce a custom media cache(`_AccumulatingMediaCache`) to patch `remove_segment_chunks` function logic. Why we cannot just have some list in `bulk_context` with segment keys that should be removed, delete here segment chunks only when `bulk_context` is None and call it also from `TaskValidationLayoutWriteSerializer._update_frames_in_bulk`?

Reply:
We can, fur sure. I just implemented it this way. In the current implementation the media cache api is the same with and without `bulk_context`.

Reply:
I guess I still have doubts regarding this, @SpecLad, could you please provide your opinion on using such approach?

Reply:
I think I agree with Maria here. The current approach with subclassing `MediaCache` seems unnecessarily complex. I'd also argue that it violates LSP, since the overridden version of `remove_segment_chunks` doesn't actually remove the chunks.

My suggestion would be to rewrite this block as something like:

```
if bulk_context:
    bulk_context.mark_segment_chunks_for_removal(chunks_to_be_removed)
else:
    media_cache = MediaCache().remove_segment_chunks(chunks_to_be_removed)
```

and in `_update_frames_in_bulk`, replace

```
bulk_context.media_cache.execute_remove_segment_chunks()
```

with

```
bulk_context.remove_marked_segment_chunks()
```

Then you can merge the methods from `_AccumulatingMediaCache` into `_TaskValidationLayoutBulkUpdateContext`.

Reply:
Ok, I removed the class.